### PR TITLE
Lint `paste(deparse(x), collapse)` in favor of `deparse1()`

### LINKED ISSUE #2615: Lint `deparse(substitute(x))` for deparse1(substitute(x))`
Since R 4.0.0 the pattern `deparse1(substitute(x))` seems to be recommended over `deparse(substitute(x))`. See the PR for examples: https://bugs.r-project.org/show_bug.cgi?id=17671 but this most likely will have many false positives.

### PR DESCRIPTION
**Pull request by emmanuel-ferdman**
### PR Summary
This PR adds `deparse1_linter()`, which flags `paste(deparse(x), collapse = ...)` and recommends `deparse1(x)` instead.

Fixes #2615 
Ref #3037

### CODE REVIEW

**Comment by MichaelChirico** [comment]
WDYT about including this as another rule under `paste_linter()`?

**Comment by MichaelChirico** [comment]
LMK if you'd like this to be included in the pending release (I'll probably submit this weekend)

**Comment by emmanuel-ferdman** [comment]
Sorry, was off for a while :) suggestion sounds good to me. I moved the check into `paste_linter()` as a new rule and dropped `deparse1_linter()`, hope that's what you had in mind :)

File: tests/testthat/test-paste_linter.R [hunk 1]
```diff
@@ -308,3 +308,62 @@ test_that("paste0(collapse=...) cases interacting with other rules are handled",
...
+  linter <- paste_linter()
+
+  expect_no_lint("deparse1(x)", linter)
+  expect_no_lint("deparse(x)", linter)
+  # no collapse=
+  expect_no_lint("paste(deparse(x))", linter)
+  # multiple positional args to paste
+  expect_no_lint("paste('Error: ', deparse(x), collapse = ' ')", linter)
+  # 'deparse' as a symbol, not a call
+  expect_no_lint("paste(deparse, collapse = ' ')", linter)
```
Comment (MichaelChirico) [inline tests/testthat/test-paste_linter.R:322]: `paste(deparse(x), collapse = ' ', sep = '.')` could also lint.

OTOH, probably we should just lint that regardless of `deparse()` as a separate `paste_linter()` rule.

Comment (MichaelChirico) [reply tests/testthat/test-paste_linter.R:322]: #3077

File: R/paste_linter.R
Comment (MichaelChirico) [inline R/paste_linter.R:11]: One edge case concern is if someone does `paste(deparse(x), collapse=',')` which will generate two lints (one for `toString()`, the other for `deparse1()`). But I _think_ we don't see that at all in practice, so I would ignore the possibility.

https://github.com/search?q=lang%3AR+%2Fdeparse1%5C%28%5B%5Cw.%5D%2B%28%3A%3A%5B%5Cw.%5D%2B%29%3F%28%5C%28%5B%5E%29%5D%2B%5C%29%29%3F%2C%5Cs*collapse%5Cs*%3D%5Cs*%5B%27%22%5D%28%5B%5E+%27%22%3B%5C%5C%5D%7C%5C%5C%5B%5En%5D%29%2F&type=code

**Review by MichaelChirico — APPROVED** [review]
Thanks!
